contracts/HubPool.sol

```diff
@@ -201,48 +209,101 @@ contract HubPool is Testable, Lockable, MultiCaller, Ownable {
         );
     }
 
     function executeRelayerRefund(
         uint256 relayerRefundRequestId,
-        uint256 leafId,
-        uint256 repaymentChainId,
-        address[] memory l1TokenAddress,
-        uint256[] memory accumulatedLpFees,
-        uint256[] memory netSendAmounts,
-        bytes32[] memory inclusionProof
-    ) public {}
+        MerkleLib.PoolRebalance memory poolRebalance,
+        bytes32[] memory proof
+    ) public {
+        RelayerRefundRequest storage relayerRefund = relayerRefundRequests[relayerRefundRequestId];
+        require(getCurrentTime() > relayerRefund.requestExpirationTimestamp, "Not passed liveness");
+
+        // Verify the leafId in the poolRebalance has not yet been claimed.
+        require(!MerkleLib.isClaimed(relayerRefund.claimedBitMap, poolRebalance.leafId), "Already claimed");
+
+        // Verify the props provided generate a leaf that, along with the proof, are included in the merkle root.
+        require(MerkleLib.verifyPoolRebalance(relayerRefund.poolRebalanceRoot, poolRebalance, proof), "Bad Proof");
 
-    function disputeRelayerRefund() public {}
+        // Set the leafId in the claimed bitmap.
+        MerkleLib.setClaimed(relayerRefund.claimedBitMap, poolRebalance.leafId);
+
+        // Decrement the unclaimedPoolRebalanceLeafs.
+        relayerRefund.unclaimedPoolRebalanceLeafs--;
+
+        // Transfer the bondAmount to back to the proposer, if this was not done before for this refund bundle.
+        if (!relayerRefund.proposerBondRepaid) {
+            relayerRefund.proposerBondRepaid = true;
+            bondToken.safeTransfer(relayerRefund.proposer, bondAmount);
+        }
+
+        // TODO call into canonical bridge to send PoolRebalance.netSendAmount for the associated
+        // PoolRebalance.tokenAddresses, to the target PoolRebalance.chainId. this will likely happen within a
+        // x_Messenger contract for each chain. these messengers will be registered in a separate process that will follow
+        // in a later PR.
+        // TODO: modify the associated utilized and pending reserves for each token sent.
+
+        emit RelayerRefundExecuted(relayerRefundRequestId, poolRebalance, msg.sender);
+    }
+
+    function disputeRelayerRefund(uint256 relayerRefundRequestId) public {
+        RelayerRefundRequest storage relayerRefund = relayerRefundRequests[relayerRefundRequestId];
+        require(
+            getCurrentTime() > relayerRefundRequests[relayerRefundRequestId].requestExpirationTimestamp,
+            "Passed liveness"
+        );
+
+        // Delete the last element in the relayerRefundRequests array. This acts to throw out the request.
+        emit RelayerRefundDisputed(relayerRefundRequestId, msg.sender);
+
+        relayerRefundRequests.pop();
+
+        // TODO: pull bonds. request price from OO.
+    }
+
+    function getNumberOfRelayRefunds() public view returns (uint256) {
+        if (relayerRefundRequests.length == 0) return 0;
```

Comment:
pretty sure you can just `return relayerRefundRequests.length` here and the current implementation is incorrect